hibernate-core/src/main/java/org/hibernate/dialect/identity/SQLiteDialectIdentityColumnSupport.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.dialect.identity;
+
+public class SQLiteDialectIdentityColumnSupport extends IdentityColumnSupportImpl {
+	@Override
+	public boolean supportsIdentityColumns() {
+		return true;
+	}
+
+  /*
+	public boolean supportsInsertSelectIdentity() {
+    return true; // As specified in NHibernate dialect
+  }
+  */
+
+	@Override
+	public boolean hasDataTypeInIdentityColumn() {
+		// As specified in NHibernate dialect
+		// FIXME true
```

Comment:
How should this be fixed in future. Is this also something related to NHibernate?